libraries/rush-lib/scripts/copyEmptyModules.js

```diff
@@ -50,12 +50,12 @@ module.exports = {
       const folderItems = await FileSystem.readFolderItemsAsync(
         relativeFolderPath ? `${jsInFolderPath}/${relativeFolderPath}` : jsInFolderPath
       );
-      for (const folderItem of folderItems) {
+      await Promise.all(folderItems.map(async (folderItem) => {
```

Comment:
We should probably use `Async.forEachAsync(...)` for this instead of Promise.all. Then we can limit the concurrency of the search (normally I believe we limit file ops to 10)

Reply:
> We should probably use `Async.forEachAsync(...)` for this instead of Promise.all. Then we can limit the concurrency of the search (normally I believe we limit file ops to 10)

Could you please clarify if I should simply replace `Promise.all` with `Async.forEachAsync(...)` in the code? It would help me understand your suggestion better. Thanks @D4N14L!

Reply:
@sanjaiyan-dev For the portion of code you are touching, it performs file operations concurrently. Using `Async.forEachAsync` _instead of_ `Promise.all` will have a similar impact on performance, while allowing us to limit the amount of concurrency when performing the file operations.

Reply:
> @sanjaiyan-dev For the portion of code you are touching, it performs file operations concurrently. Using `Async.forEachAsync` _instead of_ `Promise.all` will have a similar impact on performance, while allowing us to limit the amount of concurrency when performing the file operations.

I understand the point you're making. Are the current code suggestions correct ?

```suggestion
      await Async.forEachAsync(folderItems.map(async (folderItem) => {
```

Reply:
Not quite. Make the changes locally and you'll see the signature for Async.forEachAsync is slightly different (first param is an iterator of the objects you want to iterate, the second parameter is an async method used to perform the work)

Reply:
> Not quite. Make the changes locally and you'll see the signature for Async.forEachAsync is slightly different (first param is an iterator of the objects you want to iterate, the second parameter is an async method used to perform the work)

Sure, I will update you regarding this ASAP :)

Reply:
@D4N14L Is this correct now ? I hope so.

Reply:
@sanjaiyan-dev almost! I added a couple comments and reasoning behind them. Once those are resolved this can be merged.